noether_tpp/src/tool_path_modifiers/organization_modifiers.cpp

```diff
@@ -57,4 +59,175 @@ ToolPaths SnakeOrganizationModifier::modify(ToolPaths tool_paths) const
+      Eigen::Vector3d ref;
+      if (segment_order.empty())
+      {
+        // Use the reference start position if no segments have been ordered yet
+        ref = start_reference;
+      }
+      else
+      {
+        // Otherwise use the position of the last waypoint in the last ordered segment
+        ref = path.at(segment_order.back()).back().translation();
+      }
+
+      // Find the segment whose first waypoint is closest to the reference position
+      auto comp = std::bind(compare_distances, std::placeholders::_1, std::placeholders::_2, ref);
+      std::sort(remaining_segment_indices.begin(), remaining_segment_indices.end(), comp);
+
+      // Add the index of that segment to the output variable
+      segment_order.push_back(remaining_segment_indices.front());
+    }
+  }
+
+  return segment_order;
+}
+
+/**
+ * @brief Helper function for creating a re-ordered copy of a vector
+ */
+template <typename T>
+static std::vector<T> reorder(const std::vector<T>& input, const std::vector<std::size_t>& order)
+{
+  std::vector<T> output;
+  output.reserve(input.size());
+
+  for (std::size_t idx : order)
+  {
+    output.push_back(input.at(idx));
+  }
+
+  return output;
+}
+
+/**
+ * @brief Organizes the container of tool paths according to the length of all the tool path segments in the tool path
+ * @details The tool paths are sorted from longest to shortest. The organization of the tool path segments within each
+ * tool path and the waypoints within each tool path segment are not changed. The primary intent of this modifier is for
+ * sorting edge tool paths by length.
+ */
+static ToolPaths sortToolPathsByLength(const ToolPaths& tool_paths)
+{
+  std::vector<double> path_lengths;
+  path_lengths.reserve(tool_paths.size());
+
+  for (const ToolPath& path : tool_paths)
+  {
+    double path_length = 0.0;
+
+    for (const ToolPathSegment& segment : path)
+    {
+      for (std::size_t i = 0; i < segment.size() - 1; ++i)
+      {
+        const ToolPathWaypoint& first = segment.at(i);
+        const ToolPathWaypoint& second = segment.at(i + 1);
+        path_length += (first.inverse() * second).translation().norm();
+      }
+    }
+
+    path_lengths.push_back(path_length);
+  }
+
+  // Argsort the path lengths from largest to shortest
+  std::vector<std::size_t> order(path_lengths.size());
+  std::iota(order.begin(), order.end(), 0);
+  std::sort(order.begin(), order.end(), [&path_lengths](const std::size_t& a, const std::size_t& b) {
+    return path_lengths.at(a) > path_lengths.at(b);
+  });
+
+  // Copy the tool paths in the new order
+  return reorder(tool_paths, order);
+}
+
+static Eigen::Vector3d getAverageXAxis(const ToolPathSegment& segment)
+{
+  Eigen::Vector3d mean = Eigen::Vector3d::Zero();
+  for (const ToolPathWaypoint& w : segment)
+  {
+    mean += w.matrix().col(0).head<3>();
+  }
+  return mean / static_cast<double>(segment.size());
```

Comment:
We should probably guard against divide-by-zero here.  I would have it return Eigen::Vector3d::Zero(), but you may prefer to throw an exception.  Either way, we should do something, because nobody likes crashes when a hidden function anonymously crashes you without any chance to step in.

Edit: Looks like it will return either +inf or NaN.  Either way, probably something we want to specify the behavior of.

Reply:
Per our discussion earlier, I will just get rid of this function because its only use is in sorting the order of waypoints, which is a task that we decided the planner itself should do